### ISSUE TITLE
Changes discarded without warning when clicking off a tab

### ISSUE THREAD

**Issue by VictorioBerra** [usebruno/bruno#1997]
Open a new script, add some changes, click off the script, click back, CHANGES GONE!!!

Same with environments: open configure environment. Create an environment "prod", add some variables, create a new environment "test", click back to "prod" CHANGES GONE!!!

No warnings, no prompts, nothing. Really frustrating.

**Comment by tlaloc911** [comment]
It has happend to me

**Comment by AngelaSYuan** [comment]
Hi, I tried following the instructions @VictorioBerra wrote in the original comment, but it seemed like the variable names were saving fine. Is there another way to reproduce the bug?

**Comment by VictorioBerra** [comment]
@AngelaSYuan I will upload a video tonight or tomorrow

**Comment by AngelaSYuan** [comment]
By the way, I would like to take on this bug! For context, this would be part of a school project that has a near deadline.

Also @VictorioBerra, I seemed to be able to replicate the bug for environment variables: when modifying variables, switching environments without clicking the "Save" button loses progress. The scripts part seems to work fine still though, but I will look out for your video.

I'm thinking of adding a short text warning near the blue Save button, and preventing the user from switching environments before saving. That way, their changes aren't unexpectedly lost. Would be open to hearing others' thoughts!

**Comment by VictorioBerra** [comment]
@AngelaSYuan Yes the environment variables bug is correct. Regarding the save dialog, there is already a warning dialog, I would replicate it exactly to maintain consistency throughout Bruno:

![image](https://github.com/usebruno/bruno/assets/2934507/e66e8650-1dcb-4350-acbf-1315beeab610)

To see this dialog, open a saved request, edit it (like headers or something), and then click close without saving.

Regarding the other bug:

![Bruno_0ySRZJHwDF](https://github.com/usebruno/bruno/assets/2934507/6ed75dbf-f5fd-49e8-aec8-9cf551146f0e)

**Comment by AngelaSYuan** [comment]
Got it! For some reason, I don't see a "New Script" option in the choices.

<img width="218" alt="Screen Shot 2024-04-07 at 10 39 43 PM" src="https://github.com/usebruno/bruno/assets/86133477/78600782-c3bc-47c3-a07f-edef439f4f13">

However, I believe I fixed the bug for the environment variables bug! I can open a PR for that.

**Comment by VictorioBerra** [comment]
@AngelaSYuan That is probably a Golden Edition only feature.

**Comment by AngelaSYuan** [comment]
@VictorioBerra Right, I just wanted to ask that actually. Is there a way I download the Golden Edition for free as a developer, or is that not possible?

**Comment by sanjai0py** [comment]
Closing this issue as the [PR](https://github.com/usebruno/bruno/pull/2034) adding the feature has already been merged into Bruno!
